csharp/autobuilder/Semmle.Autobuild.Shared/Autobuilder.cs

```diff
@@ -370,6 +370,20 @@ void exitCallback(int ret, string msg, bool silent)
             }
         });
 
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                (diagnostics as IDisposable)?.Dispose();
```

Comment:
Maybe let `IDiagnosticsWriter` implement `IDisposable` (the we should also get code quality warnings if the upcast `DiagnosticsStream` -> `IDiagnosticsWriter` hides a missing disposal). It will require adding a dummy dispose in one the test class, but that should be it.

Reply:
Maybe we could argue that the naming of the class and interface could be changed a bit (scope creep).
I think we should rename `DiagnosticsStream` to `DiagnosticsWriter` and let `IDiagnosticsWriter` implement `IDisposable`.
My guess is that the only reason the interface was introduced was to allow stubbing and it seems very specific to the writer/serialize usecase.